### ISSUE TITLE
Reflection Parameter class method deprecated on PHP8

### ISSUE THREAD

**Issue by Thavarshan** [emberfuse/framework#14]
Tests fail on PHP8 because an error is thrown - `ErrorException: Method ReflectionParameter::getClass() is deprecated` on PHP8.

**Comment by Thavarshan** [comment]
Error was in `Emberfuse\Container\DependencyResolver::class` on `resolve` method where the given dependency was type checked to see whether it was a class or a primitive type.

```php
$resolution = is_null($dependency->getClass())
                ? $this->resolvePrimitive($dependency)
                : $this->resolveClass($dependency);
```

**Comment by Thavarshan** [comment]
Issue resolved in pull request #15.
